File: server/hotrod/src/test/scala/org/infinispan/server/hotrod/HotRodMarshallingTest.scala
package org.infinispan.server.hotrod

import java.util.Collections

import org.testng.annotations.Test
import org.testng.Assert._
import org.infinispan.commands.remote.ClusteredGetCommand
import org.infinispan.server.core.AbstractMarshallingTest
import org.infinispan.commons.api.BasicCacheContainer
import org.infinispan.commons.equivalence.ByteArrayEquivalence

/**
 * Tests marshalling of Hot Rod classes.
 *
 * @author Galder Zamarreño
 * @since 4.1
 */
@Test(groups = Array("functional"), testName = "server.hotrod.HotRodMarshallingTest")
class HotRodMarshallingTest extends AbstractMarshallingTest {

   def testMarshallingBigByteArrayKey() {
      val cacheKey = getBigByteArray
      val bytes = marshaller.objectToByteBuffer(cacheKey)
      val readKey = marshaller.objectFromByteBuffer(bytes).asInstanceOf[Array[Byte]]
      assertEquals(readKey, cacheKey)
   }

   def testMarshallingCommandWithBigByteArrayKey() {
      val cacheKey = getBigByteArray
      val command = new ClusteredGetCommand(cacheKey,
         BasicCacheContainer.DEFAULT_CACHE_NAME, Collections.emptySet(), false, null,
         ByteArrayEquivalence.INSTANCE)
      val bytes = marshaller.objectToByteBuffer(command)
      val readCommand = marshaller.objectFromByteBuffer(bytes).asInstanceOf[ClusteredGetCommand]
      assertEquals(readCommand, command)
   }

}

File: server/hotrod/src/test/java/org/infinispan/server/hotrod/HotRodMarshallingTest.java
package org.infinispan.server.hotrod;

import static org.testng.AssertJUnit.assertEquals;

import org.infinispan.commands.remote.ClusteredGetCommand;
import org.infinispan.commons.marshall.WrappedByteArray;
import org.infinispan.commons.util.EnumUtil;
import org.infinispan.server.core.AbstractMarshallingTest;
import org.infinispan.util.ByteString;
import org.testng.annotations.Test;

/**
 * Tests marshalling of Hot Rod classes.
 *
 * @author Galder Zamarreño
 * @since 4.1
 */
@Test(groups = "functional", testName = "server.hotrod.HotRodMarshallingTest")
public class HotRodMarshallingTest extends AbstractMarshallingTest {

   public void testMarshallingBigByteArrayKey() throws Exception {
      byte[] cacheKey = getBigByteArray();
      byte[] bytes = marshaller.objectToByteBuffer(cacheKey);
      byte[] readKey = (byte[]) marshaller.objectFromByteBuffer(bytes);
      assertEquals(readKey, cacheKey);
   }

   public void testMarshallingCommandWithBigByteArrayKey() throws Exception {
      byte[] cacheKey = getBigByteArray();
      ClusteredGetCommand command =
            new ClusteredGetCommand(new WrappedByteArray(cacheKey), ByteString.fromString("c"), 0, EnumUtil.EMPTY_BIT_SET);
      byte[] bytes = marshaller.objectToByteBuffer(command);
      ClusteredGetCommand readCommand = (ClusteredGetCommand) marshaller.objectFromByteBuffer(bytes);
      assertEquals(readCommand, command);
   }

}
